packages/opencensus-core/src/internal/util.ts

```diff
@@ -33,3 +33,44 @@ const spanRandomBuffer = randomFillSync ?
 export function randomSpanId() {
   return spanRandomBuffer().toString('hex');
 }
+
+/**
+ * Common methods to encode and decode varints and varlongs into ByteBuffers
+ * and arrays.
+ */
+export class VarInt {
+  static varIntSize(i: number): number {
+    let result = 0;
+    do {
+      result++;
+      i >>>= 7;
```

Comment:
Could you add comments for what this and the other algorithms below are doing? And maybe also link to an authoritative source for them?

Also, were these algorithms designed for 64-bit ints and do we know they work OK with doubles (JS `number`)? Maybe be sure to test this for very large numbers? (You may be able to use Big Int depending on your Node version target, or a 64-bit int polyfill)